main.rb

```diff
@@ -0,0 +1,120 @@
+# rubocop: disable Style/CaseEquality
+module Enumerable
+  UNDEFINED = Object.new
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a
+    x = 0
+    while x < arr.length
+      yield(arr[x])
+      x += 1
+    end
+    arr
+  end
```

Comment:
## my_each

- [ ] when a block is given when self is a range returns the range itself after calling the given block once for each element in self

```Ruby
range.my_each(&block)
```

_expected: 5..50
got: [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]_